### PR TITLE
Database structure

### PR DESCRIPTION
**Pull request by steven6282**
Updated the database structure similar to specified in issue #2 

Did the bare minimum required to update existing code to work with database restructure.  Still needs a bit of work, but in it's current state we can add / remove members and rfid tokens for members, and have them authenticated through a webrequest.

Supports multiple RFID tokens per member (the interface for adding them is pretty ugly atm hehe).

Currently member start_date, expiration_date, and admin flag are ignored.  Will address these in a future branch.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -23,7 +24,7 @@ def get(self,permPath):
         cn = sqlite3.connect('makermon.db')
         c = cn.cursor()
         fob_id = paths[1]
-        sql = 'SELECT * FROM members WHERE fob_id = "%s"' % ( fob_id )
+        sql = 'SELECT * FROM members m, rfid_tokens r WHERE m.enabled = 1 AND r.member_id = m.id AND r.enabled = 1 AND r.token = "%s"' % ( fob_id )
```
Comment (DallasEpperson) [inline main.py]: My only problem with this is that it leaves the door open for sql injection. Imagine if fob_id came in as something hostile (like `1"; DROP TABLE members; --`).  
  
Take a look at [the execute() method which takes two params](https://docs.python.org/2/library/sqlite3.html). That should solve it.

File: members.py [hunk 2]
```diff
@@ -35,5 +44,14 @@ def add_member(data):
     cn.commit()
 
 def delete_member(member_id):
-    cn.execute('DELETE FROM members WHERE ROWID = ?', member_id)
+    cn.execute('UPDATE members SET enabled = 0 WHERE id = ?', member_id)
```
Comment (DallasEpperson) [inline members.py:47]: Here's an example of that two-param .execute() method I was talking about. If you use this one in main.py, we should be good :)

**Review by DallasEpperson — CHANGES_REQUESTED** [review]
Looks 99% ready to me, just take a look at that sql injection vector.

File: main.py (hunk 1 above)
Comment (steven6282) [reply main.py]: That makes sense.  I was wondering what the two different syntax was for, I just had not looked it up yet.  I knew SQL injection was a concern, didn't know that paramatizing the execute call took care of it automatically.

How do I fix this in a pull request on GitHub.  Do I have to cancel the pull request update the branch and then make another pull request?  Or is there some way to update the pull request automatically when updating the branch?

Comment (DallasEpperson) [reply main.py]: It's the latter - when you push your next commit to this branch, the pull request will be updated with the latest and greatest.

File: members.py (hunk 2 above)
Comment (steven6282) [reply members.py:47]: Hmm, ok so the commit I just did automatically updated the other one, but the note you left here seems to still flag it as having a change requested.  What do we need to do about this one?

**Review by DallasEpperson — APPROVED** [review]

File: members.py (hunk 2 above)
Comment (DallasEpperson) [reply members.py:47]: That just means no one had reviewed the changes yet. I just reviewed (and approved). This branch is ready to be merged in now, which I'll do in just a moment.

**Comment by DallasEpperson** [comment]
Something you might want to consider adding as an upcoming feature is some data integrity validation before actually storing to the DB. In sqlite, a column's specified data type isn't actually actually honored. E.g. you can specify that a column should be INTEGER, but you can still store strings in it.

**Comment by DallasEpperson** [comment]
You might also want to consider making the rfid_tokens.token value unique in the schema. All just suggestions for a future update, though. Merging now!

**Pull request merged**